ostd/src/mm/page/mod.rs

```diff
@@ -126,8 +126,8 @@ impl<M: PageMeta> Page<M> {
     /// The physical address must represent a valid page and the caller must already hold one
     /// reference count.
     pub(in crate::mm) unsafe fn inc_ref_count(paddr: Paddr) {
-        let page = unsafe { ManuallyDrop::new(Self::from_raw(paddr)) };
-        let _page = page.clone();
+        let vaddr: Vaddr = mapping::page_to_meta::<PagingConsts>(paddr);
+        unsafe { &(*(vaddr as *const MetaSlot)).ref_count }.fetch_add(1, Ordering::Relaxed);
```

Comment:
Can you help add some safety comments to distinguish the safety requirement of this function and the satisfied requirement of the operations in the unsafe block? The original code base is not well documented either. We can discuss about how to improve.

Since the function is harmless if the physical address is valid and the caller holds a count, the unsafe block should be justified. We need to ensure that the virtual address points to initialized metadata slot. You can debug assert it and comment it with safety comments:

```rust
        debug_assert(paddr < MAX_PADDR)
        let vaddr: Vaddr = mapping::page_to_meta::<PagingConsts>(paddr);
        // SAFETY: the physicall address is valid, so the virtual address points to an initialized metadata slot.
        unsafe { &(*(vaddr as *const MetaSlot)).ref_count }.fetch_add(1, Ordering::Relaxed);
```